spring-kafka/README.md

```diff
@@ -0,0 +1,357 @@
+third parameter accepts a `com.google.protobuf.MessageLite` because we used our `KafkaProtobufSerializer` for the value.
+
+## Consumer
+
+The consumer part of the messages follows some of the patterns used in the producer but adds a layer of complexity because
+we may have different strategies to read messages from the Kafka brokers. To consume messages from Kafka topics the following 
+blueprint definitions should be used:
+
+```xml
+<type-converters>
+  <bean class="org.hitachivantara.osgi.service.blueprint.GenericsTypeConverter">
+    <argument>
+      <map key-type="java.lang.Class" value-type="java.lang.Class">
+        <entry key="org.springframework.kafka.core.DefaultKafkaConsumerFactory" value="org.springframework.kafka.core.ConsumerFactory" />
+        <entry key="org.apache.kafka.common.serialization.StringDeserializer" value="org.apache.kafka.common.serialization.Deserializer" />
+        <entry key="org.hitachivantara.spring.kafka.protobuf.KafkaProtobufDeserializer" value="org.apache.kafka.common.serialization.Deserializer" />
+        <entry key="com.hitachivantara.protobuf.api.proto.SimpleMessage$1" value="com.google.protobuf.Parser" />
+        <entry key="com.hitachivantara.protobuf.api.proto.ComplexMessage$1" value="com.google.protobuf.Parser" />
+      </map>
+    </argument>
+  </bean>
+</type-converters>
+
+<!-- Consumer properties -->
+<bean id="kafkaConsumerProperties" class="java.util.HashMap">
+  <argument>
+    <map key-type="java.lang.String" value-type="java.lang.Object">
+      <entry key="bootstrap.servers" value="localhost:9092"/>
+      <entry key="group.id" value="Default_Consumer_Group_ID"/>
+      <entry key="enable.auto.commit" value="false"/>
+      <entry key="session.timeout.ms" value="15000"/>
+    </map>
+  </argument>
+</bean>
+
+<!-- Simple Message Listener-->
+<bean id="kafkaSimpleMessageListener" class="com.hitachivantara.kafka.SimpleListener"
+      init-method="start" destroy-method="stop" >
+  <property name="consumerFactory"> <!-- type converter used on this argument -->
+    <bean class="org.springframework.kafka.core.DefaultKafkaConsumerFactory">
+      <argument ref="kafkaConsumerProperties"/>
+      <argument> <!-- type converter used on this argument -->
+        <bean class="org.apache.kafka.common.serialization.StringDeserializer" />
+      </argument>
+      <argument> <!-- type converter used on this argument -->
+        <bean class="org.hitachivantara.spring.kafka.protobuf.KafkaProtobufDeserializer">
+          <argument> <!-- type converter used on this argument -->
+            <bean class="com.hitachivantara.protobuf.api.proto.SimpleMessage" factory-method="parser"/>
+          </argument>
+        </bean>
+      </argument>
+    </bean>
+  </property>
+  <property name="topic" value="SimpleTopic"/>
+  <property name="groupId" value="SimpleConsumerGroupId"/>
+</bean>
+
```

Comment:
Like I previously mentioned, we should separate simple and complex examples a bit more to make it easier to process the information.